rinex/src/context/mod.rs

```diff
@@ -24,561 +22,268 @@ use rinex_qc_traits::HtmlReport;
-                    Ok(_) => trace!("loaded \"{}\"", filename),
-                    Err(e) => error!("failed to load \"{}\", {:?}", path.display(), e),
-                }
-            }
+impl<'a> BlobData<'a> {
+    /// Returns reference to inner RINEX data.
+    pub fn as_rinex(&self) -> Option<&'a Rinex> {
+        match self {
+            Self::Rinex(r) => Some(r),
+            _ => None,
         }
-        Ok(ret)
     }
-    /// Load individual file into Context
-    pub fn load(&mut self, path: &PathBuf) -> Result<(), Error> {
-        let fullpath = path.to_string_lossy().to_string();
-        let filename = path
-            .file_name()
-            .ok_or(Error::FileNameDetermination)?
-            .to_string_lossy()
-            .to_string();
-        if let Ok(rnx) = Rinex::from_path(path) {
-            if rnx.is_observation_rinex() {
-                self.load_obs(path, &rnx)?;
-                trace!("loaded observations \"{}\"", filename);
-            } else if rnx.is_navigation_rinex() {
-                self.load_nav(path, &rnx)?;
-                trace!("loaded brdc nav \"{}\"", filename);
-            } else if rnx.is_meteo_rinex() {
-                self.load_meteo(path, &rnx)?;
-                trace!("loaded meteo observations \"{}\"", filename);
-            } else if rnx.is_clock_rinex() {
-                self.load_clock(path, &rnx)?;
-                trace!("loaded clock data \"{}\"", filename);
-            } else if rnx.is_ionex() {
-                self.load_ionex(path, &rnx)?;
-                trace!("loaded ionex \"{}\"", filename);
-            } else if rnx.is_antex() {
-                self.load_antex(path, &rnx)?;
-                trace!("loaded antex dataset \"{}\"", filename);
-            } else {
-                return Err(Error::NonSupportedType);
-            }
-        } else if let Ok(sp3) = SP3::from_file(&fullpath) {
-            self.load_sp3(path, &sp3)?;
-            trace!("loaded sp3 \"{}\"", filename);
-        }
-        Ok(())
-    }
-    /// Unwraps inner RINEX data, by preference order:
-    /// 1. Observation Data if provided
-    /// 2. Navigation Data if provided
-    /// 3. Meteo Data if provided
-    /// 4. Ionex Data if provided
-    /// 5. ATX Data if provided
-    fn provided_rinex(&self) -> Option<&ProvidedData<Rinex>> {
-        if let Some(data) = &self.obs {
-            Some(data)
-        } else if let Some(data) = &self.nav {
-            Some(data)
-        } else if let Some(data) = &self.meteo {
-            Some(data)
-        } else if let Some(data) = &self.ionex {
-            Some(data)
-        } else if let Some(data) = &self.atx {
-            Some(data)
-        } else if let Some(data) = &self.clk {
-            Some(data)
-        } else {
-            None
+    /// Returns mutable reference to inner RINEX data.
+    pub fn as_mut_rinex(&mut self) -> Option<&'a mut Rinex> {
+        match self {
+            Self::Rinex(r) => Some(r),
```

Comment:
@lnicola 

But in processing workflow, you sometimes need mutable access, for example to rework the context prior some algorithm. This method is trying to provide such access, but it fails to compile.
That's because `r` here has an actual `&& Rinex` type I think.

How do you solve this ? I think we should use some kind of fancy structure, like `RefCell<Rinex>` in BlobData and not simply `&Rinex`